impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -200,6 +207,84 @@ Rect MakeReferenceUVs(const Rect& reference, const Rect& rect) {
 int ScaleBlurRadius(Scalar radius, Scalar scalar) {
   return static_cast<int>(std::round(radius * scalar));
 }
+
+class GeometryExtractor : public TypedContentsVisitor<ColorSourceContents> {
```

Comment:
This approach feels really fragile. We should have the geometry when we construct the mask blur, right?

Reply:
Yea, in our current implementation we do.  There is also a bug in our current implementation where it seems to only work for ColorSourceContents, which isn't what we want.

It isn't as fragile as it seems since the compiler is supporting it.  If another subclass of ColorSourceContents is introduced, the visitor will have a compile error were you just implement the standard Visitor pattern and it will work.

The way this could break is if we introduce a new Contents that has a geometry that isn't a ColorSourceContents.  But this way is probably more robust to that change than relying on having a ColorSourceContents when we create the blur.

Reply:
Also, passing in the geometry as part of the filterinput and as a separate parameter can create the opportunity for them to differ which would be a logical error.

Reply:
> we don't need to worry about mask blurs on arbitrary filers

What's enforcing that?  Why doesn't our API enforce that?  I worry that trying to ignore what the API allows creates oversights like with what happened with mask blur filters on textures.  That's the downside to doing it the other way.

Reply:
Tests enforce it, same as it ever was!

Reply:
@bdero I chatted with jonah and what we decided to do is just reduce the scope of the Visitor so that it only goes as far as ColorSourceContents.  That's all I need to implement what I want and cuts down a lot of the code.  If we need an more expansive visitor in the future we can implement that similarly to what I've done here.

I wish I understood the wizardry that std::variant uses to implement their visitor since it would resolve this issue.

Reply:
I think you might also be able to simplify the visitor further too, if you're limiting the scope to just the work needed to land this change. In that case, you don't even need to know what you are visiting, just whether or not it has a geometry you can use.

Reply:
Okay @jonahwilliams, please take a look, I removed all the subclasses of `ColorSourceContents` and `FilterContents`.  I don't need to know anything below that to implement what we need.